src/Metadata/PropertyTypeIterable.php

```diff
@@ -17,52 +17,73 @@ final class PropertyTypeIterable extends PropertyTypeArray
     /**
      * @var string
      */
-    private $collectionClass;
+    private $traversableClass;
 
     /**
-     * @param class-string<\Traversable>|null $collectionClass
+     * @param class-string<\Traversable>|null $traversableClass
      */
-    public function __construct(PropertyType $subType, bool $hashmap, bool $nullable, string $collectionClass = null)
+    public function __construct(PropertyType $subType, bool $hashmap, bool $nullable, string $traversableClass = null)
     {
-        parent::__construct($subType, $hashmap, $nullable, null != $collectionClass);
+        parent::__construct($subType, $hashmap, $nullable, null != $traversableClass);
 
-        $this->collectionClass = $collectionClass;
+        $this->traversableClass = $traversableClass;
     }
 
     public function __toString(): string
     {
         if ($this->subType instanceof PropertyTypeUnknown) {
-            return 'array'.($this->isCollection() ? '|\\'.$this->collectionClass : '');
+            return 'array'.($this->isTraversable() ? '|\\'.$this->traversableClass : '');
         }
 
         $array = $this->isHashmap() ? '[string]' : '[]';
-        if ($this->isCollection()) {
+        if ($this->isTraversable()) {
             $collectionType = $this->isHashmap() ? ', string' : '';
-            $array .= sprintf('|\\%s<%s%s>', $this->collectionClass, $this->subType, $collectionType);
+            $array .= sprintf('|\\%s<%s%s>', $this->traversableClass, $this->subType, $collectionType);
         }
 
         return ((string) $this->subType).$array.AbstractPropertyType::__toString();
     }
 
+    /**
+     * @deprecated Please prefer using {@link getTraversableClass}
+     */
     public function getCollectionClass(): ?string
     {
-        return $this->collectionClass;
+        return $this->traversableClass;
     }
 
+    /**
+     * @deprecated Please prefer using {@link isTraversable}
+     */
     public function isCollection(): bool
     {
-        return null != $this->getCollectionClass();
+        return null != $this->getTraversableClass();
```

Comment:
this does not work as getTraversableClass throws an exception. we need to check the class property:

```suggestion
        return null != $this->traversableClass;
```

Reply:
yeah its a bit weird :-D 

we could also not overwrite the isCollection/getCollection methods in the PropertyTypeIterable class so it falls back on the legacy behaviour. that might be simpler. the method has already been merged but not in any release, so we can still remove it again.